logstash-core/lib/logstash/pipeline.rb

```diff
@@ -78,7 +63,7 @@ def initialize(config_str, settings = {})
     #
     # This need to be configured before we evaluate the code to make
     # sure the metric instance is correctly send to the plugin.
-    @metric = settings.fetch(:metric, Instrument::NullMetric.new)
+    @metric = Instrument::NullMetric.new
```

Comment:
👍 for this improvement.  might be would be good to add an small comment to provide context that metric it has to be initialized (if not willing to be empty) through the accessor.